E2E (Atomic): Skip the editor exit check if in mobile classic (wp-admin) view

WunderBart opened the pull request:
#### Proposed Changes

We're currently using the Classic View on our Atomic test site, which forces the wp-admin top nav in mobile view. Let's make the "Return to Calypso dashboard" check only when the Calypso nav is present. 

@Automattic/team-calypso-platform, it seems odd that when selecting the `Classic view` on the `posts` page, the wp-admin nav is forced, but **only** for the mobile viewport. The desktop viewport has the same navbar regardless of the `View` setting. What do you think? Do you know if this is intended for the mobile viewport?

| Desktop nav for both Classic and Default views |
| ------------- |
| <img width="902" alt="Screenshot 2022-12-09 at 11 43 24" src="https://user-images.githubusercontent.com/1451471/206684704-4293ad5a-e963-4b07-8c84-35fa9604f4fc.png"> |

| Mobile nav w/ Classic view | Mobile nav w/ Default view |
| ------------- | ------------- |
| <img width="484" alt="Screenshot 2022-12-09 at 11 38 24" src="https://user-images.githubusercontent.com/1451471/206684116-0c873b9b-aee6-46db-9520-94739a7ccc61.png"> | <img width="447" alt="Screenshot 2022-12-09 at 11 37 49" src="https://user-images.githubusercontent.com/1451471/206684140-9bd183b7-65bf-49f2-9890-1771147fb752.png"> |




#### Testing Instructions

The `editor__navbar.ts` suite should be passing for the [Gutenberg Atomic](https://teamcity.a8c.com/buildConfiguration/calypso_WPComTests_gutenberg_atomic_production_mobile) job.

Related: p1669253368977539/1669227776.675969-slack-CBTN58FTJ

WunderBart committed:
Skip the editor exit check if in mobile classic (wp-admin) view

worldomonation commented on test/e2e/specs/editor/editor__navbar.ts:
```diff
@@ -43,6 +43,12 @@ describe( DataHelper.createSuiteTitle( `Editor: Navbar` ), function () {
 	it( 'Return to Calypso dashboard', async function () {
-		await editorPage.exitEditor();
+		const isMobileClassicView =
+			envVariables.VIEWPORT_NAME === 'mobile' && ( await page.isVisible( '#wpadminbar' ) );
```
**issue:blocking**: I would not use `page.isVisible()` to reliably check whether an element is actually visible. The [official API](https://playwright.dev/docs/api/class-page#page-is-visible) isn't too clear on this but [this snippet](https://playwright.dev/docs/api/class-page#page-is-visible-option-timeout), combined with my past experience using `page.isXYZ` methods suggests that it should not be used.

Can I suggest the alternative `locator.isVisible` instead?

WunderBart replied:
Thanks for the review, @worldomonation!

> Can I suggest the alternative locator.isVisible instead?

Sure, `locator` sounds good! Do you recall what the specific issue with `page.isVisible` was, though? Seems like both `page.isVisible` and `locator.isVisible` have the `timeout` option marked as `DEPRECATED` (regarding the snippet you mentioned). 🤔  I'm wondering if there's any difference between those two implementation-wise since the logic seems identical.

worldomonation replied:
> Seems like both page.isVisible and locator.isVisible have the timeout option marked as DEPRECATED

Hmm, that is a little unexpected. I recall the Playwright team mentioning somewhere public that the `page.isXYZ` methods are no longer considered good use and to use the Locator version instead, but the suggestion may have been actually to use `locator.waitFor` instead.

I went searching for information on `page.isXYZ` methods and whether it's good practice. I found https://github.com/microsoft/playwright/issues/12672 which in turn references https://github.com/microsoft/playwright/issues/12224 as to why the`timeout` field was deprecated, but doesn't make statement on whether it's good practice to use these methods. There's also a recent question https://github.com/microsoft/playwright/issues/19396.

I suppose I should amend my suggestion to say `locator.waitFor` instead. Thanks for pointing it out.

WunderBart committed:
Use locator.isVisible instead page.isVIsible
As per https://github.com/Automattic/wp-calypso/pull/70982#discussion_r1044921906

WunderBart committed:
Extend comment

griffbrad commented:
@alshakero Thought I’d reach out about the discrepancy between desktop and mobile viewports in the Classic vs Defaults views as described by @WunderBart above because I know Vertex folks have worked in that space relatively recently while trying to ensure Help Center is available in different contexts. Do you know the motivation behind that discrepancy? Is there any reason we shouldn’t offer a consistent experience between those viewport sizes and views?

fullofcaffeine reviewed (approved):
This testing well for me locally! 

```
VIEWPORT_NAME=mobile TEST_ON_ATOMIC=true  node ../../node_modules/.bin/jest specs/editor/editor__navbar.ts

  console.log
    TestAccount:gutenbergAtomicSiteUser => Logging in via Login Page

      at TestAccount.log (../../packages/calypso-e2e/src/lib/test-account.ts:198:12)

 PASS  specs/editor/editor__navbar.ts (10.489 s)
  Editor: Navbar
    ✓ Go to the new post page (6935 ms)
    ✓ Return to Calypso dashboard (14 ms)

Test Suites: 1 passed, 1 total
Tests:       2 passed, 2 total
Snapshots:   0 total
Time:        10.631 s, estimated 24 s
```

```
VIEWPORT_NAME=desktop TEST_ON_ATOMIC=true  node ../../node_modules/.bin/jest specs/editor/editor__navbar.ts
  console.log
    TestAccount:gutenbergAtomicSiteUser => Logging in via Login Page

      at TestAccount.log (../../packages/calypso-e2e/src/lib/test-account.ts:198:12)

 PASS  specs/editor/editor__navbar.ts (11.841 s)
  Editor: Navbar
    ✓ Go to the new post page (5869 ms)
    ✓ Return to Calypso dashboard (2713 ms)

Test Suites: 1 passed, 1 total
Tests:       2 passed, 2 total
Snapshots:   0 total
Time:        11.971 s
Ran all test suites matching /specs\/editor\/editor__navbar.ts/i.
```

I think the failures on CI are due to flaky tests, as this change seems pretty isolated. I just restarted them, but pre-emptively accepting this as it seems to work great!

fullofcaffeine commented:
Rebasing to see if will fix the failing tests.

WunderBart commented:
I've moved the nav/viewport issue to https://github.com/Automattic/wp-calypso/issues/71376. I'll merge this and follow up once we know what to do with the navigation discrepancy.
